src/actions/gpxActions.js

```diff
@@ -1,11 +1,26 @@
 import axios from "axios";
 import * as types from "./actionTypes";
 import * as api from "../util/api.js"
+import * as gpsUtil from "../util/gps";
 
 function receiveGPXTracks(data) {
+    const array = [];
+    let currentDistance = 0.0;
+    let lastElement  = null;
+    data.map(element => {
+        let distance = 0.0;
+        if (lastElement === null) {
+            distance = 0.0;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Review this useless assignment: "distance" already holds the assigned value along all execution paths. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS4165)